scripts/automation/Radius/Functions/Private/settings/Update-JCRUsersJson.ps1

```diff
@@ -0,0 +1,72 @@
+function Update-JCRUsersJson {
+    [CmdletBinding()]
+    param (
+        [Parameter()]
+        [switch]
+        $force
+    )
+    begin {
+    }
+    process {
+        # validate that the system association data is correct in users.json:
+        # $userArray = Get-UserJsonData
+        foreach ($user in $Global:JCRRadiusMembers) {
+            $MatchedUser = $GLOBAL:JCRUsers[$user.userID]
+            $userArrayObject, $userIndex = Get-UserFromTable -userID $user.userID -jsonFilePath "$JCScriptRoot/users.json"
+
+            if ($userIndex -ge 0) {
+                # $userArrayObject
+                # write-host "$($userArrayObject.username) | $($userArrayObject.userId)"
+                $currentSystemObject = $userArrayObject.systemAssociations | Select-Object systemId, hostname, osFamily
+                $incomingSystemObject = $Global:JCRAssociations[$userArrayObject.userId].systemAssociations
+                $incomingList = New-Object System.Collections.ArrayList
+                foreach ($system in $Global:JCRAssociations[$userArrayObject.userId].systemAssociations) {
+                    $incomingList.Add(
+                        [pscustomobject]@{
+                            systemId = $system.systemId
+                            hostname = $system.hostname
+                            osFamily = $system.osFamily
+                        }) | Out-Null
+                }
+
+                # determine if there's some difference that needs to be recorded:
+                try {
+                    if ($currentSystemObject -eq $null) {
+                        $difference = $incomingList
+                        Set-UserTable -index $userIndex -username $MatchedUser.username -localUsername $MatchedUser.systemUsername -systemAssociationsObject ($incomingList)
+                    } else {
+                        # write-host "test for differences"
+                        if ($currentSystemObject -eq $incomingList) {
+                            # write-host "nothing to do"
+                        } else {
+                            # write-host "writing user to do"
+                            Set-UserTable -index $userIndex -username $MatchedUser.username -localUsername $MatchedUser.systemUsername -systemAssociationsObject ($incomingList)
+                        }
+                    }
+                } catch {
+                    <#Do this if a terminating exception happens#>
```

Comment:
Should we add debug or error message here?

Reply:
In this case, we just need to determine if there are changes to the user to system association table, we setting the `$difference` variable null here to note that there are no changes. No need to throw a message for the end user.